### PR TITLE
Update the way we publish

### PR DESCRIPTION
**Pull request by vazexqi**
### What does this PR do?

Node 8 makes it more difficult to `require` global libraries. So the previous way of trying to use the globally installed shelljs library doesn't work. See https://nodejs.org/api/modules.html#modules_loading_from_the_global_folders for the full resolution order.

We could make the old way work but that would require us to either install it in the $HOME directory, or mess with NODE_PATH. Neither sounds desirable.

So, instead the main changes are:

1. You do a `npm install` at the top-level. This brings in the dependencies.
1. When we do a `git clean` we ignore the top-level node_module which only contains things like shelljs, lerna, etc. These are never bundled so they don't affect the generated vsix.

### CODE REVIEW

File: docs/publishing.md [hunk 1]
```diff
@@ -31,8 +31,7 @@ It is possible to run each step manually as illustrated below.
 
 ### Prerequisite
 
-* Lerna is properly installed (`npm install -g lerna@2.0.0`).
-* shelljs is properly installed (`npm install -g shelljs`).
```
Comment (vazexqi) [inline docs/publishing.md]: You cannot require shelljs globally using 

```
const shell = require('shelljs');
```

so we remove this line.

File: scripts/publish.js [hunk 2]
```diff
@@ -99,9 +99,6 @@ shell.exec(
   `lerna publish --force-publish --exact --repo-version ${nextVersion} --yes --skip-npm`
 );
 
-// Reformat with prettier since lerna changes the formatting in package.json
-shell.exec('./scripts/reformat-with-prettier.js');
```
Comment (vazexqi) [inline scripts/publish.js]: We tie this now in the top-level package.json to the end of the vscode:package.

@DatGuyJonathan

File: scripts/publish.js [hunk 3]
```diff
@@ -84,7 +84,7 @@ if (checkTags.includes(nextVersion)) {
 }
 
 // Real-clean
-shell.exec('git clean -xfd');
+shell.exec('git clean -xfd -e node_modules');
```
Comment (vazexqi) [inline scripts/publish.js]: So when we clean we now leave the top-level node_modules (with shelljs) alone.

File: docs/publishing.md [hunk 4]
```diff
@@ -25,16 +25,18 @@ The files under scripts use [shelljs/shx](https://github.com/shelljs/shx) and
 [shelljs/shelljs](https://github.com/shelljs/shelljs) to write scripts in a
 portable manner across platforms.
 
+1. `git checkout -t origin release/vxx.yy.zz`
```
Comment (vazexqi) [inline docs/publishing.md]: I think Winston never updated this doc with the simplified steps.
But here it is.

File: scripts/publish.js (hunk 2 above)
Comment (DatGuyJonathan) [reply scripts/publish.js]: My top level package.json also got formatted during packaging, so I think the reformat script needs to be edited to include top level package.json.

File: docs/publishing.md [hunk 5]
```diff
@@ -25,16 +25,18 @@ The files under scripts use [shelljs/shx](https://github.com/shelljs/shx) and
...
 It is possible to run each step manually as illustrated below.
 
 ## Packaging as .vsix
 
 ### Prerequisite
 
-* Lerna is properly installed (`npm install -g lerna@2.0.0`).
-* shelljs is properly installed (`npm install -g shelljs`).
-* You have a clean workspace. You can use `git clean -xfd` to get to a pristine
-  stage.
```
Comment (lcampos) [inline docs/publishing.md]: Shouldn't we keep this but modify it to leave top-level node_modules (git clean -xfd -e node_modules) ?

File: scripts/publish.js (hunk 2 above)
Comment (vazexqi) [reply scripts/publish.js]: Interesting. Sure I can include that.

File: docs/publishing.md (hunk 5 above)
Comment (vazexqi) [reply docs/publishing.md]: The publish scripts actually runs `git clean -xfd -e node_modules` for you. That's why I avoided doing it here.

**Pull request merged**